selfdrive/ui/ui.cc

```diff
@@ -403,35 +452,82 @@ void handle_message(UIState *s, Message * msg) {
-    s->scene.uilayout_sidebarcollapsed = datad.sidebarCollapsed;
-    s->scene.uilayout_mapenabled = datad.mapEnabled;
-
-    bool hasSidebar = !s->scene.uilayout_sidebarcollapsed;
-    bool mapEnabled = s->scene.uilayout_mapenabled;
-    if (mapEnabled) {
-      s->scene.ui_viz_rx = hasSidebar ? (box_x+nav_w) : (box_x+nav_w-(bdr_s*4));
-      s->scene.ui_viz_rw = hasSidebar ? (box_w-nav_w) : (box_w-nav_w+(bdr_s*4));
-      s->scene.ui_viz_ro = -(sbr_w + 4*bdr_s);
-    } else {
-      s->scene.ui_viz_rx = hasSidebar ? box_x : (box_x-sbr_w+bdr_s*2);
-      s->scene.ui_viz_rw = hasSidebar ? box_w : (box_w+sbr_w-(bdr_s*2));
-      s->scene.ui_viz_ro = hasSidebar ? -(sbr_w - 6*bdr_s) : 0;
-    }
   } else if (eventd.which == cereal_Event_liveMapData) {
     struct cereal_LiveMapData datad;
     cereal_read_LiveMapData(&datad, eventd.liveMapData);
     s->scene.map_valid = datad.mapValid;
+  } else if (eventd.which == cereal_Event_thermal) {
+    struct cereal_ThermalData datad;
+    cereal_read_ThermalData(&datad, eventd.thermal);
+
+    if (datad.networkType == cereal_ThermalData_NetworkType_none) {
+      s->scene.networkType = NETWORKTYPE_NONE;
+    } else if (datad.networkType == cereal_ThermalData_NetworkType_wifi) {
+      s->scene.networkType = NETWORKTYPE_WIFI;
+    } else if (datad.networkType == cereal_ThermalData_NetworkType_cell2G) {
+      s->scene.networkType = NETWORKTYPE_CELL2G;
+    } else if (datad.networkType == cereal_ThermalData_NetworkType_cell3G) {
+      s->scene.networkType = NETWORKTYPE_CELL3G;
+    } else if (datad.networkType == cereal_ThermalData_NetworkType_cell4G) {
+      s->scene.networkType = NETWORKTYPE_CELL4G;
+    } else if (datad.networkType == cereal_ThermalData_NetworkType_cell5G) {
+      s->scene.networkType = NETWORKTYPE_CELL5G;
+    }
+
+    s->scene.batteryPercent = datad.batteryPercent;
+    snprintf(s->scene.batteryStatus, sizeof(s->scene.batteryStatus), "%s", datad.batteryStatus.str);
+    s->scene.freeSpace = datad.freeSpace;
+    s->scene.thermalStatus = datad.thermalStatus;
+    s->scene.paTemp = datad.pa0;
+  } else if (eventd.which == cereal_Event_ubloxGnss) {
+    struct cereal_UbloxGnss datad;
+    cereal_read_UbloxGnss(&datad, eventd.ubloxGnss);
+    struct cereal_UbloxGnss_MeasurementReport reportdatad;
+    cereal_read_UbloxGnss_MeasurementReport(&reportdatad, datad.measurementReport);
+
+    s->scene.satelliteCount = reportdatad.numMeas;
+  } else if (eventd.which == cereal_Event_health) {
+    struct cereal_HealthData datad;
+    cereal_read_HealthData(&datad, eventd.health);
+
+    s->scene.hwType = datad.hwType;
+    s->hardware_timeout = 5*30; // 5 seconds at 30 fps
   }
   capn_free(&ctx);
 }
 
+static void check_messages(UIState *s) {
+  while(true) {
+    auto polls = s->poller->poll(0);
+
+    if (polls.size() == 0)
+      break;
+
+    for (auto sock : polls){
+      Message * msg = sock->receive();
+      if (msg == NULL) continue;
+
+      if (s->vision_connected) {
```

Comment:
Why is this here?

Reply:
Maybe have the check_message function return if something was received, and use that to set awake?